src/overlays/actors/ovl_Bg_Spot08_Iceblock/z_bg_spot08_iceblock.c

```diff
@@ -26,47 +34,420 @@ const ActorInit Bg_Spot08_Iceblock_InitVars = {
     (ActorFunc)BgSpot08Iceblock_Update,
     (ActorFunc)BgSpot08Iceblock_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot08_Iceblock/func_808B0960.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot08_Iceblock/func_808B0968.s")
+extern Gfx D_06000DE0[];           // Large iceberg spawned in place of Jabu
+extern CollisionHeader D_06001904; // Large iceberg spawned in place of Jabu
+extern Gfx D_06002BD0[];           // Hexagonal ice floe
+extern CollisionHeader D_06002FD8; // Hexagonal ice floe
+
+void BgSpot08Iceblock_SetupAction(BgSpot08Iceblock* this, BgSpot08IceblockActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+void BgSpot08Iceblock_InitDynaPoly(BgSpot08Iceblock* this, GlobalContext* globalCtx, CollisionHeader* colHeader,
+                                   DynaPolyMoveFlag flags) {
+    s32 pad;
+    CollisionHeader* dest;
+    s32 pad2;
+
+    dest = NULL;
```

Comment:
we usually call this colHeader and the arg as `collision`
that may change at a later time but might as well make it consistent with the other init dynapoly funcs